### ISSUE TITLE
Node Sass does not yet support your current environment: Windows 64-bit with Unsupported runtime (57)

### ISSUE THREAD

**Issue by SetCodesToFire** [jakearchibald/wittr#20]
I faced this issue when I used the latest LTS version of node.js i.e, 8.9.0 and npm version 5.5.1

I resolved this by updating the gulp-saas version number to 3.1.0 in the package.json file and it resolved, I wish you change the same :)

**Comment by s-slavchev** [comment]
I also noticed that many of you are having trouble with `npm install`. I made small research and found out that the used **gulp-sass version "^2.0.4"** in the **devDependencies** has a dependency of **node-sass : "^3.4.2"**  which can be satisfied by version 3.13.1.  But version 3.13.1 supports only node versions till 7 [release v3.13.1](https://github.com/sass/node-sass/releases/tag/v3.13.1) 
The easiest solution to the problem, which will save a lot of trouble for windows users with node versions 8 and 9, is to bump gulp-sass version to the latest "^3.1.0". It requires the latest node-sass [release v4.6.0](https://github.com/sass/node-sass/releases/tag/v4.6.0). 

This way we avoid the need for building the module from source. 

Note: There will be some warnings in the initial install, because of unmet optional dependencies. 

**Edit:** The commit which fixes the problem was merged to the master branch and the issue #21 was closed.

**Comment by Giorat** [comment]
@s-slavchev what commands should I use to fix this? Or do I need to modify package.json? I'm on x64 too and I really want to solve this.

**Comment by SetCodesToFire** [comment]
@s-slavchev you just open your package.json file and in that you will see gulp-saas in the dev dependencies :::

Step 1:

"gulp-sass": "^2.0.4"

change this to 

"gulp-sass": "^3.1.0"

step 2: now run 
                            $ npm install 
in the command prompt


done.



Now everytime you git hard reset,(according to the lecture and move to the other branches.) you will not have to npm install again and again and hence you are done with these two steps. 👍

**Comment by Giorat** [comment]
@SetCodesToFire thanks now it's working!

**Comment by jakearchibald** [comment]
Should be working now!

**Comment by mohit-wadhwa** [comment]
Updating the gulp-saas version number to 3.1.0 solved the issue and works fine.
Thanks

**Comment by hmrowie** [comment]
Updating the gulp-saas version number to 3.1.0 and run npm update (or if you are using yarn run "yarn upgrade" and problem solve)

**Comment by SassNinja** [comment]
Updating to gulp-sass 3.1.0 solved the issue for me, too
Thanks

**Comment by nardocesar** [comment]
Still working. Upgraded to gulp-sass 3.1.0 and solve my problem. Thanks!

**Comment by tan-ahmed** [comment]
I spent 2 hours of updating/installing; Node, Npm, gulp, gulp-sass etc on my windows machine... nothing worked!! Till I found this and tried it. JESUS FREAKING CHRIST, YOU ARE A LIFE SAVER!

**Comment by cliu86** [comment]
You are LIFE saver !   THanks

**Comment by AdhamRammadan** [comment]
running  `npm rebuild node-sass ` also helps.

**Comment by ghost** [comment]
This absolutely solved my problem. Trying to update my gulp-Sass!

**Comment by leftblank0** [comment]
And, if you get the following error follow this -

"Could not load the Visual C++ component VCBuild.exe”

https://stackoverflow.com/questions/21658832/npm-install-error-msb3428-could-not-load-the-visual-c-component-vcbuild-ex

**Comment by pallanagendra** [comment]
running npm rebuild node-sass  helped me... thanks a lot

**Comment by phil123456** [comment]
I dont have any gulp-sass, jsut sass and still have the error

**Comment by JonV8** [comment]
thank you, OMG this saved my ass haha

**Comment by JMKP** [comment]
please help me i changed on package.json to the v3.1.0 but i am still getting error

Starting 'sass'...
[20:32:45] 'sass' errored after 167 ms
[20:32:45] Error: Node Sass does not yet support your current environment: Windows 64-bit with Unsupported runtime (57)
For more information on which environments are supported please see:
https://github.com/sass/node-sass/releases/tag/v3.13.1

**Comment by rchawla501** [comment]
@SetCodesToFire thanks now it's working!

**Comment by fabiobaser** [comment]
@JMKP `npm rebuild node-sass` didn't work for me. I ran `which node-sass` and found out that node-sass was also installed via brew. I just deleted the file `which node-sass` was pointing to and reinstalled it via `npm install -g node-sass`. Hope that helps

**Comment by Titoratus** [comment]
This error occurs when I try run Gulp in directory of my project that cloned by SmartGit. For other project's directories Gulp works fine! And `npm rebuild node-sass` and `npm install -g node-sass`didn't work for me. BUT `npm install node-sass` (without -g) helped!

**Comment by lakscode** [comment]
@Titoratus  I was also facing the same issue. Thank you for the solution. "npm install node-sass" worked for me

**Comment by WouterVanherck** [comment]
You could also get this when you recently installed node.js on a fresh computer. Before you can update your node-sass, the following commands should be executed:

1.	npm --add-python-to-path='true' --debug install --global windows-build-tools
2.	npm config set msvs_version 2015
3.	npm i node-sass@latest

**Comment by jrswenson** [comment]
@WouterVanherck, 

What does line 2 do?  What do you set it as if you're using VS2017 or VS Code?

**Comment by WouterVanherck** [comment]
@jrswenson,  
Actually, besides the obvious which is setting the version, I'm not 100% sure what it does.  
I just know that I had a generic error and Googled my way to a solution.  
Execute the first command and check if that's enough, because that alone might fix it already.  

As for the VS2017/VS Code question:  
That command helped me fix my issue, even if I'm using VS2017 an VS Code too.  
Thinking about it, might as well do `--msvs_version=2017` instead.

Let me know if it works out and good luck coding!

**Comment by liushuang** [comment]
npm install node-sass works for me.Thanks a lot~

**Comment by yashwp** [comment]
[0] Module build failed: Error: Node Sass does not yet support your current environment: Windows 64-bit with Unsupported runtime (64)
[0] For more information on which environments are supported please see:
[0] https://github.com/sass/node-sass/releases/tag/v4.3.0


I don't have gulp-sass in my package.json how do I fix this?
Please help.
Node version - 10.5.0
node-sass - 4.3.0

**Comment by voloKh28** [comment]
try to delete `node_modules` and `node-sass` from `package.json dependencies`

and after that run `npm install node-sass`

**Comment by nathiyakumar** [comment]
npm install node-sass works for me.Thanks a lot~

**Comment by Luanitntu** [comment]
you should run: npm i node-sass --save-dev once again

**Comment by kronosindia** [comment]
sdf

**Comment by Dorf** [comment]
here's what worked for me on macOS:

Downgrade node from "11.0.0 Current" to "8.12.0 LTS", (https://nodejs.org/en/)

following these instructions to downgrade:
https://apple.stackexchange.com/questions/171530/how-do-i-downgrade-node-or-install-a-specific-previous-version-using-homebrew#answer-262119

then `npm rebuild node-sass`

and it worked

**Comment by johnsmithsren** [comment]
> @Titoratus I was also facing the same issue. Thank you for the solution. "npm install node-sass" worked for me

worked for me too

**Comment by arasrasouli** [comment]
I did the steps which are here:
https://github.com/dlmanning/gulp-sass/wiki/Update-to-the-latest-Gulp-Sass

then it worked for me

**Comment by GTOsss** [comment]
This error appears on new versions of Node js (> 8.0.0 ). Switching to version 8 solves the problem. 
In my case ```nvm use 8.12.0```

**Comment by ocordero** [comment]
The error happens when you have the latest version of node js. The solution was to update the project with npm update.

**Comment by PavelZgurski** [comment]
npm rebuild node-sass thanx.

**Comment by johnniefp** [comment]
gulp-sass update worked like a treat.
Thanks!

**Comment by Alexandre1BR** [comment]
> This error occurs when I try run Gulp in directory of my project that cloned by SmartGit. For other project's directories Gulp works fine! And `npm rebuild node-sass` and `npm install -g node-sass`didn't work for me. BUT `npm install node-sass` (without -g) helped!

npm install node-sass works for me.Thanks a lot~

@antonioribeiro

**Comment by Andrew-Dyachenko** [comment]
> Updating to gulp-sass 3.1.0 solved the issue for me, too
> Thanks

Confirmed - it helped relatively to ZURB Emails: `"gulp-sass": "^2.1.0" => "gulp-sass": "^4.0.0"`

**Comment by ychernetsov** [comment]
`npm rebuild node-sass` didn't help
helped:
`npm uninstall node-sass`
`npm install node-sass`

**Comment by mahajanabhij** [comment]
> here's what worked for me on macOS:
> 
> Downgrade node from "11.0.0 Current" to "8.12.0 LTS", (https://nodejs.org/en/)
> 
> following these instructions to downgrade:
> https://apple.stackexchange.com/questions/171530/how-do-i-downgrade-node-or-install-a-specific-previous-version-using-homebrew#answer-262119
> 
> then `npm rebuild node-sass`
> 
> and it worked

Worked for me on windows too.

**Comment by tarcisioreis** [comment]
I resolve my problem make this tip down:

Tip for windows users, uninstall node-sass and reinstall (npm install node-sass) and then build (npm build) and run the application (npm serve -open)

**Comment by LiamKarlMitchell** [comment]
Fixing with an Ionic angular application.

Node v12.2.0 Windows 10.

Build failed with error code: 1 but I ignored this as it later said "Binary is fine"
```
npm uninstall node-sass
npm uninstall node-sass -g
npm install node-sass
npm build
ionic serve
```

**Comment by surendra246** [comment]
Please run sudo npm rebuild node-sass from terminal and please don't forgot the sudo before using command.

Above command resolved my issue!!

**Comment by dungphanxuan** [comment]
I have same problem, so i update to "gulp-sass": "^3.1.0", so i got error 
`TypeError [ERR_INVALID_CALLBACK]: Callback must be a function
    at makeCallback (fs.js:136:11)
    at Object.utimes (fs.js:1094:14)
    at C:\xampp\htdocs\github\dungphanxuan\yiiframework.com\node_modules\gulp-touch\index.js:11:13
    at wrappedMapper (C:\xampp\htdocs\github\dungphanxuan\yiiframework.com\node_modules\map-stream\index.js:84:19)
    at Stream.stream.write (C:\xampp\htdocs\github\dungphanxuan\yiiframework.com\node_modules\map-stream\index.js:96:21)
    at Pumpify.ondata (C:\xampp\htdocs\github\dungphanxuan\yiiframework.com\node_modules\readable-stream\lib\_stream_readable.js:619
:20)
    at Pumpify.emit (events.js:198:13)
    at Pumpify.EventEmitter.emit (domain.js:466:23)
    at addChunk (C:\xampp\htdocs\github\dungphanxuan\yiiframework.com\node_modules\readable-stream\lib\_stream_readable.js:291:12)
    at readableAddChunk (C:\xampp\htdocs\github\dungphanxuan\yiiframework.com\node_modules\readable-stream\lib\_stream_readable.js:2
78:11)
`

**Comment by shubham-99fusion** [comment]
> This error occurs when I try run Gulp in directory of my project that cloned by SmartGit. For other project's directories Gulp works fine! And `npm rebuild node-sass` and `npm install -g node-sass`didn't work for me. BUT `npm install node-sass` (without -g) helped!


**npm install node-sass**  worked!!!

**Comment by sandimilohanic** [comment]
When i used the node version 13.1. i had the error "Node Sass does not yet support your current environment: Windows 64-bit with Unsupported runtime"

When i installed the version  < 13 the problem has fixed.

**Comment by gabrielbarbosaTolentinos** [comment]
> you should run: npm i node-sass --save-dev once again


good people still exist in this world

**Comment by feitianhouren** [comment]
After I update node.js to 13.2.0, I suffer this error["Node Sass does not yet support your current environment: Windows 64-bit with Unsupported runtime (75) ***not same as 57 in this question title], but I don't remember the last vesion of node.js, so I uninstall node.js 13.2.0, try other versions. At last, I go back to node.js 10.17.0 version, I solve this problem.

**Comment by A1rPun** [comment]
@LiamKarlMitchell That did the trick for me :+1:.
I have `node-sass` as  devDependency so I used this:

```
npm uninstall node-sass
npm uninstall node-sass -g
npm install node-sass -D
```

**Comment by augustoproiete** [comment]
I got the same error after updating Node.js to v13.3.0.

Updating the package `node-sass` resolved it.

```
npm i node-sass --save-dev
```

---

![image](https://user-images.githubusercontent.com/177608/70345373-4ac3d180-1832-11ea-8d27-4f40e5b3806e.png)

```
Failed to compile
./src/styles/index.scss (./node_modules/css-loader/dist/cjs.js??ref--6-oneOf-5-1!./node_modules/postcss-loader/src??postcss!./node_modules/resolve-url-loader??ref--6-oneOf-5-3!./node_modules/sass-loader/lib/loader.js??ref--6-oneOf-5-4!./src/styles/index.scss)
Error: Node Sass does not yet support your current environment: Windows 64-bit with Unsupported runtime (79)
For more information on which environments are supported please see:
https://github.com/sass/node-sass/releases/tag/v4.12.0
This error occurred during the build time and cannot be dismissed.
```

**Comment by rcouture27** [comment]
> `npm rebuild node-sass` didn't help
> helped:
> `npm uninstall node-sass`
> `npm install node-sass`

This is the only thing that worked on MacOS for angular 6 app

**Comment by 22krishan** [comment]
npm update
npm install
npm start

update command updates to latest node-sass version.
Above mentioned steps worked for me.

**Comment by rohit-aneja** [comment]
`npm install -g node-sass`

This command saved the day for me

**Comment by sali141** [comment]
> running `npm rebuild node-sass ` also helps.

this worked fine with me. thanks @rohit-aneja

**Comment by jchilela** [comment]
if the command "which node-sass" is returning empty, you have to install node-sass
Just run: 

npm install node-sass

Still working in May 2020

**Comment by sunnytambi** [comment]
For me, I had to update node-sass-chokidar from 1.3.4 to 1.4.0 and the build started working.

**Comment by juliemoura** [comment]
@Titoratus Thank you bro, I spend one hour on this sh*t. You are a really lifesaver.

**Comment by juliemoura** [comment]
@Titoratus Thank you bro, i've spend one hour on this sh*t. You are a really lifesaver.

**Comment by chuxuantinh** [comment]
NodeJS  Supported node-sass version *
Node    15        5.0+   
Node    14        4.14+ 
Node    13        4.13+, < 5.0   
Node    12        4.12+  
Node    11        4.10+, < 5.0   
Node    10        4.9+   
Node     8        4.5.3+, < 5.0  
Node  <  8   <    5.0     < 57
13

Check file package.json
"node-sass": "^4.14.1",
    "preboot": "^4.5.2",
    "primeng": "^4.3.0",
    "raw-loader": "^0.5.1",
    "read-appsettings-json": "^1.0.98",

=> In my situation problem was in the version of node. As I used the newest version 16.6.4 but the version of node-sass was old (4.14.1), I downgraded node to 14.15.5 (nvm install 14.15.5), ran npm install node-sass@4.14.1, rebuilt node npm rebuild node-sass and just then started my app npm start. Worked!

Steps:

1.nvm install 14.15.5 (downgrade node to version that satisfies node-sass)
*https://nodejs.org/en/blog/release/v14.15.5
2.nvm use 14.15.5 
3.npm install node-sass@4.14.1 (reinstall node-sass)
4.npm rebuild node-sass
5.npm start

After executing these commands everything is working fine for me.
Chu Xuân Tình
